picard/ui/__init__.py

```diff
@@ -19,10 +19,39 @@
 
 
 import uuid
+from picard import config
 from PyQt5 import QtCore, QtWidgets
+from picard.util import restore_method
 
 
-class PicardDialog(QtWidgets.QDialog):
+class PreserveGeometry:
+
+    defaultsize = None
+    autorestore = True
+
+    def __init__(self):
+        config.Option("persist", self.opt_name(), QtCore.QByteArray())
+        if self.autorestore:
+            self.restore_geometry()
+        if getattr(self, 'finished', None):
+            self.finished.connect(self.save_geometry)
+
+    def opt_name(self):
```

Comment:
@property?

Reply:
I'm not fond of `@property`, it hides the fact we call a function, and since function call overhead in Python is quite high, i prefer not. Plus this one is only used locally.